src/table-semantic/styled-components.js

```diff
@@ -48,59 +48,72 @@ export const StyledTableHeadRow = styled<{}>('tr', ({$theme}) => {
   return {};
 });
 
-export const StyledTableHeadCell = styled<{}>('th', ({$theme}) => {
-  return {
-    ...$theme.typography.font350,
-    position: 'sticky',
-    top: 0,
-    paddingTop: $theme.sizing.scale500,
-    paddingRight: $theme.sizing.scale600,
-    paddingBottom: $theme.sizing.scale500,
-    paddingLeft: $theme.sizing.scale600,
-    backgroundColor: $theme.colors.tableHeadBackgroundColor,
-    color: $theme.colors.contentPrimary,
-    textAlign: 'left',
-    verticalAlign: 'top',
-    whiteSpace: 'nowrap',
-
-    // We have to use pseudo elements to add the border for headers
-    // because browsers don't properly handle borders on sticky cells.
-    // The cells stay fixed in place, but the borders scroll.
-    '::before': {
-      content: '""',
-      position: 'absolute',
-      top: '0',
-      right: '100%',
-      bottom: '0',
-      borderLeftColor: $theme.borders.border300.borderColor,
-      borderLeftStyle: $theme.borders.border300.borderStyle,
-      borderLeftWidth: $theme.borders.border300.borderWidth,
-    },
+type StyledTableHeadCellPropsT = {
+  $col?: {},
```

Comment:
@chasestarr Would be happy to have input on how to make these `{}` types more specific.

Reply:
if there's a generic param that defaults to `{}`, can flow infer when a defined type is provided in `$col`?

Reply:
I would love to do that, but I can't figure out how to use generics and `styled<>` together.

```jsx
type StyledCompPropsT<T> = {
  col: T,
};

const StyledComp = styled<StyledCompPropsT<T>>('div', () => ({}));
// Flow error: Cannot resolve name `T`.
```